proto/remote_execution.proto

```diff
@@ -2294,9 +2294,17 @@ message VMMetadata {
     // ExecuteResponse. The digest function used to compute the digest should
     // match the one associated with the invocation.
     Digest execute_response_digest = 4;
+
+    // A unique UUID assigned each time a snapshot is used.
+    // For example, even if the same execution is retried multiple times using
+    // the same snapshot key, each run will have a unqiue snapshot_id.
+    string snapshot_id = 5;
```

Comment:
I am a bit unclear on why a task has a snapshot ID associated with it - the snapshot ID is a property of the VM, not the task right?

Reply:
Let's say you have an invocation link for a workflow. This workflow (VM B) was forked from VM A. In my mental model, if you grab snapshot ID B from that invocation page, you're starting from the snapshot VM B saved. If you want to start from the snapshot VM A saved, you should follow the VM metadata link to the previous workflow link and grab snapshot ID A from there.

Under that model, my intention is that this snapshot_id (set for `last_executed_task`) represents the ID of the snapshot the VM was resumed from (snapshot-id-A). 

The snapshot ID set directly on the VMMetadata object (line 2308) represents the ID of the current snapshot run (before and during task execution - snapshot-id-B). 

I think we can make this a little more clear by adding some more context to the UI. "VM resumed from snapshot ID X" "VM saved to snapshot ID Y". Wdyt?

Reply:
Ack - will do in a follow up